### PR TITLE
Footer

### LINKED ISSUE #13: Footer
- [ ] Create a new component in shared-ui for footer. This component should take in a list of objects where each object contains 1) the name of the link and 2) where the link directs. This is to make our component flexible so we can use it on the mainsite if we want to pass in a different set of footer strings. 
- [ ] Import the footer into the index.tsx of main-site
- [ ] The footer should be responsive based on the desktop + mobile designs 
- [ ] Use the respective button in shared ui components for the join mailing list button (don't create your own)
- [ ] If you want to use a typography styled component (for example, H4) but it isn't exactly the right size, you can do styled(H4) and then add additional styling to override the original styling. (see example below) 

NOTE: for this ticket, don't worry about getting the functionality of the joining mailing list to work yet - just focus on styling. 

Screen Shot 2022-09-13 at 6 51 00 PM

### PR DESCRIPTION
**Pull request by deanframe**
<Summary line - One sentence describing your intended set of changes>

Fixes #13.

Changelist:

- adds a footer according to figma designs

Notes:

- I couldn't figure out a good way to pin the footer below the background image, as the background image is positioned absolute and its height changes with screen resolution and can't be measured easily.  @jzhzhang66 @karynayen any help appreciated!
- This means that for now the footer just sits below the page content, instead of below the background.

Tested:

- Tested at all resolutions

Screenshots & Screencasts:
image
image
image

### CODE REVIEW

File: shared-ui/components/footer/Footer.styles.ts [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+  flex-direction: column;
+  flex-wrap: wrap;
+  align-self: center;
+  flex-basis: 40%;
+  padding: 8em 0 8em 10em;
+`
+
+const StyledTab = styled(P)`{
+  color: ${colors.WHITE};
+  font-family: ${fonts.nunitoSansSemibold};
```
Comment (jzhzhang66) [inline shared-ui/components/footer/Footer.styles.ts]: Screen Shot 2022-09-20 at 5 52 22 PM

The text here looks a bit lighter - could you try one of the lighter nunito sans font families? 

Also on the Figma the letters look farther apart so let's add a bit of letter-spacing to imitate that!

File: shared-ui/components/footer/Footer.styles.ts [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+  display: flex;
+  flex-direction: column;
+  justify-content: space-evenly;
+  @media ${max.tablet} {
+    padding: 0;
+    width: 20em;
+  }
+  @media ${max.desktop} {
+    align-items: center;
+    gap: 12px;
```
Comment (jzhzhang66) [inline shared-ui/components/footer/Footer.styles.ts]: can we change this to em to be consistent with everything else?

File: shared-ui/components/footer/Footer.styles.ts [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+    justify-content: center;
+  }
+  padding: 3em 0;
+`
+
+const StyledEmailInput = styled.input`
```
Comment (jzhzhang66) [inline shared-ui/components/footer/Footer.styles.ts:90]: The styling for the placeholder on the Figma looks a bit different (blue, letter spacing, etc. so let's add in those design elements! 
Screen Shot 2022-09-20 at 5 57 50 PM
Screen Shot 2022-09-20 at 5 58 52 PM

File: shared-ui/components/footer/Footer.tsx [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+      <StyledButtonContainer>
+        <StyledEmailInput
+          placeholder='Enter your email'
+          value={email}
+          onChange={(e: React.FormEvent<HTMLInputElement>): void => setEmail(e.currentTarget.value)}
+        />
+        <StyledButtonRow>
+          <PrimaryButton btnText='Join our mailing list' btnLink='' />
+          <StyledSocialMediaButtons>
+            <a href="https://www.instagram.com/hackbeanpot/"><img width='41' height='41' src={Instagram} /></a>
```
Comment (jzhzhang66) [inline shared-ui/components/footer/Footer.tsx]: For all of these lets add target blank so it opens in a new tab

File: shared-ui/components/footer/Footer.styles.ts [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+  color: ${colors.HEADER_FOOTER_BLUE};
+  font-size: 1em;
+  background-color: ${colors.WHITE};
+  border: 1px solid white;
```
Comment (jzhzhang66) [inline shared-ui/components/footer/Footer.styles.ts]: Let's use em for everything!

File: shared-ui/components/footer/Footer.styles.ts [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+    ::placeholder {
+      text-align: center;
+    }
+  }
+`
+
+const StyledButtonRow = styled.div`
+  display: flex;
+  flex-direction: row;
+  gap: 12px;
```
Comment (jzhzhang66) [inline shared-ui/components/footer/Footer.styles.ts]: Let's use em for everything!

File: shared-ui/components/footer/Footer.styles.ts [hunk 7]
```diff
@@ -0,0 +1,147 @@
...
+  align-items: center;
+  @media ${max.desktop} {
+    flex-direction: column;
+  }
+`
+
+const StyledSocialMediaButtons = styled.div`
+  display: flex;
+  flex-direction: row;
+  gap: 12px;
```
Comment (jzhzhang66) [inline shared-ui/components/footer/Footer.styles.ts]: Let's use em for everything!

File: shared-ui/components/footer/Footer.tsx [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+
+const Footer: React.FC<FooterProps> = ({ tabs }) => {
+  const [email, setEmail] = useState<string>('');
+
+  return (
+    <StyledFooter>
+      <StyledHackBeanpotLogo
+        src={HackBeanpotWhiteLogo}
+        alt='White HackBeanpot Logo'
+        width='191px'
```
Comment (jzhzhang66) [inline shared-ui/components/footer/Footer.tsx]: Is it possible to use em here or not really

**Review by jzhzhang66 — CHANGES_REQUESTED** [review]
Awesome work Dean!!!!! I think it's fine that the footer isn't at the bottom right now. As we add more sections, it'll get pushed to the bottom eventually LOLLLL. 

In addition to the comments I left, at this kind of screensize could we move everything but the logo to the left a bit? Everything looks too close to the right I feel like
Screen Shot 2022-09-20 at 6 02 51 PM

File: shared-ui/components/footer/Footer.styles.ts (hunk 3 above)
Comment (jzhzhang66) [inline shared-ui/components/footer/Footer.styles.ts:90]: Can we also add a max-width of maybe like 28em? On larger screens the input looks a bit too wide LOL

File: shared-ui/components/footer/Footer.styles.ts [hunk 9]
```diff
@@ -0,0 +1,147 @@
...
+    flex-direction: row;
+    align-items: flex-start;
+  }
+`
+
+const StyledLogoContainer = styled.div`
+  flex-basis: 20%;
+`
+
+const StyledHackBeanpotLogo = styled.img`
```
Comment (jzhzhang66) [inline shared-ui/components/footer/Footer.styles.ts]: I feel like the logo looks a decent amount bigger in the design

**Review by jzhzhang66 — CHANGES_REQUESTED** [review]
Can we also add some more spacing between these elements? On the Figma they're more spaced out 
Screen Shot 2022-09-20 at 6 29 53 PM

File: shared-ui/components/footer/Footer.tsx [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+  StyledFooter,
+  StyledFooterText,
+  StyledFooterTextLink,
+  StyledFooterTextContainer,
+  StyledHackBeanpotLogo,
+  StyledSocialMediaButtons,
+  StyledTab,
+  StyledTabContainer
+} from './Footer.styles';
+import HackBeanpotWhiteLogo from '../../images/hackbeanpot-white-logo.png';
```
Comment (karynayen) [inline shared-ui/components/footer/Footer.tsx]: is it possible to use an svg for the hbp logo?

File: shared-ui/components/footer/Footer.tsx [hunk 11]
```diff
@@ -0,0 +1,69 @@
...
+      </StyledFooterTextContainer>
+      <StyledButtonContainer>
+        <StyledEmailInput
+          placeholder='Enter your email'
+          value={email}
+          onChange={(e: React.FormEvent<HTMLInputElement>): void => setEmail(e.currentTarget.value)}
+        />
+        <StyledButtonRow>
+          <PrimaryButton btnText='Join our mailing list' btnLink='' />
+          <StyledSocialMediaButtons>
```
Comment (karynayen) [inline shared-ui/components/footer/Footer.tsx]: the social media buttons look a tad bit large in my opinion. looking at the figma, it seems like the should be about 1/3 the height of the "join our mailing address" button when at full screen. 

Also, I think the font in the textbox and buttons should be Nunito Sans?

Lastly, there may need to be a little more space between the "mailing address" button and the social media buttons

current version:
Screen Shot 2022-09-20 at 8 54 49 PM

figma:
Screen Shot 2022-09-20 at 8 51 07 PM

**Review by karynayen — CHANGES_REQUESTED** [review]
GREAT JOB DEAN!!! I JUST HAVE A FEW COMMENTS!!!! BUT IT LOOKS BEAUTIFUL!!!

File: shared-ui/components/footer/Footer.tsx (hunk 11 above)
Comment (deanframe) [reply shared-ui/components/footer/Footer.tsx]: Updated global primary button style to use nunito sans

**Comment by deanframe** [comment]
@jzhzhang66 @karynayen ready for re-review!

**Review by jzhzhang66 — APPROVED** [review]
LGTM LET'S GO JOHN MAYERRRRRR

**Review by karynayen — APPROVED** [review]
LGTMT!!! GREAT JOB!!

**Pull request merged**